packages/ra-core/src/auth/usePermissions.ts

```diff
@@ -42,25 +32,23 @@ const emptyParams = {};
  *     };
  */
 const usePermissions = <Permissions = any, Error = any>(
-    params = emptyParams
-): State<Permissions, Error> => {
-    const [state, setState] = useSafeSetState<State<Permissions, Error>>({
-        isLoading: true,
-    });
+    params: UseQueryOptions<Permissions, Error> = {
+        refetchOnMount: false,
+    }
+) => {
     const getPermissions = useGetPermissions();
-    useEffect(() => {
-        getPermissions(params)
-            .then(permissions => {
-                setState({ isLoading: false, permissions });
-            })
-            .catch(error => {
-                setState({
-                    isLoading: false,
-                    error,
-                });
-            });
-    }, [getPermissions, params, setState]);
-    return state;
+
+    const { data, isLoading, error } = useQuery(
+        ['ra-Permissions'],
+        getPermissions,
+        params
+    );
+
+    return {
```

Comment:
You probably need to memo this, or this will trigger unwanted rerenders

Reply:
Are you sure about that?
It's basically just a wrapper for `useQuery`.

Reply:
Yes I'm sure: `useQuery` will return an identical object if nothing has changed, but your hook will always return a new object.

Reply:
See #8174

Reply:
done.